SmartDeviceLink/SDLSoftButtonManager.m

```diff
@@ -129,231 +117,52 @@ - (void)setSoftButtonObjects:(NSArray<SDLSoftButtonObject *> *)softButtonObjects
 
     _softButtonObjects = softButtonObjects;
 
-    [self updateWithCompletionHandler:nil];
-}
-
-- (nullable SDLSoftButtonObject *)softButtonObjectNamed:(NSString *)name {
-    for (SDLSoftButtonObject *object in self.softButtonObjects) {
-        if ([object.name isEqualToString:name]) {
-            return object;
-        }
-    }
-
-    return nil;
+    SDLSoftButtonReplaceOperation *op = [[SDLSoftButtonReplaceOperation alloc] initWithConnectionManager:self.connectionManager fileManager:self.fileManager capabilities:self.softButtonCapabilities softButtonObjects:_softButtonObjects mainField1:self.currentMainField1];
+    [self.transactionQueue cancelAllOperations];
+    [self.transactionQueue addOperation:op];
 }
 
 - (void)sdl_transitionSoftButton:(SDLSoftButtonObject *)softButton {
-    self.isDirty = YES;
-    [self updateWithCompletionHandler:nil];
+    SDLSoftButtonTransitionOperation *op = [[SDLSoftButtonTransitionOperation alloc] initWithConnectionManager:self.connectionManager capabilities:self.softButtonCapabilities softButtons:self.softButtonObjects mainField1:self.currentMainField1];
+    [self.transactionQueue addOperation:op];
```

Comment:
I don't think so. A batched update may not change the soft buttons, since batching affects both "soft button" and "text and graphic" managers. In addition, imagine you have the following transaction queue:

1. replace [s1, s2, s3]
2. transition [s1]

Then you start a batch with:
3. transition [s2]

If you cancel 1 and 2, then 3 will no longer make any sense.

Reply:
Makes sense.